src/material.rs

```diff
@@ -1,45 +1,220 @@
+//! Material parameters for mesh rendering.
+
 use render::BasicPipelineState;
 use scene::Color;
 use texture::Texture;
 
-/// Material is the enhancement of Texture that is used to setup appearance of [`Mesh`](struct.Mesh.html).
-#[allow(missing_docs)]
+pub use self::basic::Basic;
+
+/// Basic material parameters.
+pub mod basic {
+    use super::*;
+
+    /// Parameters for a basic mesh material.
+    #[derive(Clone, Debug, Default)]
+    pub struct Basic {
+        /// Solid color applied in the absense of `map`.
+        ///
+        /// Default: `0x000000` (black).
+        pub color: Color,
+
+        /// Texture applied using the mesh texture co-ordiantes.
+        ///
+        /// Default: `None`.
+        pub map: Option<Texture<[f32; 4]>>,
+
+        /// Specifies which pipeline should be used to render the mesh.
+        ///
+        /// Default: `Solid`.
+        pub pipeline: Pipeline,
+    }
+
+    /// Specifies which pipeline should be used to render the mesh.
+    #[derive(Clone, Debug)]
+    pub enum Pipeline {
+        /// Renders the mesh as a solid.
+        Solid,
```

Comment:
Perhaps this should be renamed `Fill` to mimic the implementation pipeline (`pso_mesh_basic_fill`)?